### PR TITLE
[next] add env variables for testing

### PR DESCRIPTION
**Pull request by feedthejim**
Adds code for 3 experiments:
- `VERCEL_NEXT_PRELOAD_INIT`:  preloads the page(s) route code in the init phase
- `VERCEL_NEXT_REDUCED_LAMBDA_SIZE`: reduces the size for grouping
- `VERCEL_NEXT_GROUPING_DISABLED`: disable grouping altogether

### CODE REVIEW

File: packages/next/src/utils.ts [hunk 1]
```diff
@@ -1359,7 +1359,13 @@ export type LambdaGroup = {
...
 };
 
-export const MAX_UNCOMPRESSED_LAMBDA_SIZE = 250 * MIB;
+const USE_REDUCED_LAMBDA_SIZE = !!process.env.VERCEL_NEXT_REDUCED_LAMBDA_SIZE;
+
+// the ratio for a compressed lambda is 1/4 the size of an uncompressed lambda
+export const MAX_UNCOMPRESSED_LAMBDA_SIZE =
+  (USE_REDUCED_LAMBDA_SIZE ? 50 : 250) * MIB;
+export const MAX_COMPRESSED_LAMBDA_SIZE =
+  (USE_REDUCED_LAMBDA_SIZE ? 10 : 50) * MIB;
```
Comment (ijjk) [inline packages/next/src/utils.ts:1399]: Maybe make this configurable but seems good to start

**Review by ijjk — DISMISSED** [review]

File: packages/next/src/server-build.ts [hunk 2]
```diff
@@ -913,9 +913,30 @@ export async function serverBuild({
         }
       }
 
+      let finalLauncher = group.isAppRouter ? appLauncher : launcher;
+
+      if (process.env.VERCEL_NEXT_PRELOAD_INIT) {
+        let preloadCode = `
+          require('./.next/server/pages/_app.js');
+          require('./.next/server/pages/_error.js');
```
Comment (ijjk) [inline packages/next/src/server-build.ts]:
```suggestion
```

File: packages/next/src/server-build.ts [hunk 3]
```diff
@@ -913,9 +913,29 @@ export async function serverBuild({
         }
       }
 
+      let finalLauncher = group.isAppRouter ? appLauncher : launcher;
+
+      if (process.env.VERCEL_NEXT_PRELOAD_INIT) {
+        let preloadCode = `
```
Comment (ijjk) [inline packages/next/src/server-build.ts]:
```suggestion
        let preloadCode = group.isAppRouter ? '' : `
```

**Review by ijjk — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by styfle** [comment]
Closing since Jimmy said this isn't needed anymore

**Pull request closed without merging**